src/doors.c

```diff
@@ -0,0 +1,41 @@
+#include <cub3D.h>
+
+static void	toggle_surrounding_doors(t_window *window);
+
+void	door_handler(mlx_key_data_t keydata, void *arg)
+{
+	t_window	*window;
+
+	window = (t_window *)arg;
+	if (keydata.key == MLX_KEY_SPACE && keydata.action == MLX_PRESS)
+	{
+		toggle_surrounding_doors(window);
+		draw_scene(window);
+	}
+}
+
+static void	toggle_surrounding_doors(t_window *window)
```

Comment:
this segfaults if you are close to the maps border, because it cant iterate over the empty 'outside' cubes